llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -1347,14 +1348,11 @@ bool RISCVVLOptimizer::runOnMachineFunction(MachineFunction &MF) {
   auto PushOperands = [this, &Worklist](MachineInstr &MI,
                                         bool IgnoreSameBlock) {
     for (auto &Op : MI.operands()) {
-      if (!Op.isReg() || !Op.isUse() || !Op.getReg().isVirtual())
-        continue;
-
-      if (!isVectorRegClass(Op.getReg(), MRI))
+      if (!Op.isReg() || !Op.isUse() || !Op.getReg().isVirtual() ||
+          !isVectorRegClass(Op.getReg(), MRI))
```

Comment:
We can drop this isVectorRegClass check too if we check for isCandidate below as well?

Reply:
I don't know that we check isVectorRegClass in isCandidate, although we could probably move this check to isCandidate?

Reply:
Oh sorry you're right. Ignore my comment